platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -1189,6 +1205,47 @@ private void writeLeavesToPathToKeyValue(
                 throw new UncheckedIOException(e);
             }
             statisticsUpdater.countFlushLeavesWritten();
+        }
+
+        // end writing
+        final DataFileReader pathToKeyValueReader = pathToKeyValue.endWriting();
+        statisticsUpdater.setFlushLeavesStoreFileSize(pathToKeyValueReader);
+
+        runPathToKeyStoreCompaction();
+    }
+
+    /** Write all the given leaf records to keyToPath */
+    private void writeLeavesToKeyToPath(
+            final long firstLeafPath,
+            final long lastLeafPath,
+            @NonNull final List<VirtualLeafBytes> dirtyLeaves,
+            @NonNull final List<VirtualLeafBytes> deletedLeaves,
+            boolean isReconnect)
+            throws IOException {
+        // If both streams are empty, no new data files should be created. One simple way to
+        // check emptiness is to use iterators. The iterators are consumed on a single thread
+        // (the current thread), but it still makes sense to use parallel streams as supplying
+        // elements to the stream includes expensive operations like serialization to bytes
+        final Iterator<VirtualLeafBytes> dirtyIterator = dirtyLeaves.stream()
+                .parallel()
+                .sorted(Comparator.comparingLong(VirtualLeafBytes::path))
```

Comment:
We sort `dirtyLeaves` twice.

Reply:
This is addressed in the new version of the fix